libhwmon/wrapadl.cpp

```diff
@@ -214,9 +214,10 @@ int wrap_adl_get_power_usage(wrap_adl_handle *adlh, int gpuindex, unsigned int*
 	}
 	int* power = new int;
 	rc = adlh->adl2Overdrive6CurrentPowerGet(adlh->context, adlh->phys_logi_device_id[gpuindex], 0, power);
-	double watts = adlf2double(*power);
-	unsigned int res = (unsigned int)(watts * 1000);
-	*miliwatts = res;
+	int fraction = 	(*power) & 0x00FF; //last 8 bits seem to be decimal part
```

Comment:
How about instead of all this:

*milliwatts = *power / 3.90625;

Reply:
Isn't it more efficient to use bitwise operations, shifts and divisions by powers of 2? 

I don't get the operation (sorry if I'm being slow haha) 59648 should be 233000mW... Am I right?